scala-generator/src/main/scala/models/generator/ScalaClientCommon.scala

```diff
@@ -27,6 +27,27 @@ class Client${config.asyncTypeParam(Some("Sync")).map(p => s"[$p]").getOrElse(""
 """.trim
   }
 
+  def responseEnvelopeTrait(name: String): String = {
+    s"""
+trait $name[T] {
```

Comment:
why generate a trait and a case class?

Reply:
fair question - was not sure exactly how this would be used in tests and traits are easier to mock (though don't have a concrete use case)

Reply:
I think case classes are actually easier to use in tests than traits. All you need to do is call `ResponseImpl(my, test, values)` instead of extending a trait. And since there are no methods on ResponseImpl, there's no need for any logic to be overridden.

Reply:
👍